pkgs/development/compilers/llvm/15/clang/default.nix

```diff
@@ -0,0 +1,130 @@
+{ lib, stdenv, llvm_meta
+, monorepoSrc, runCommand
+, substituteAll, cmake, ninja, libxml2, libllvm, version, python3
+, buildLlvmTools
+, fixDarwinDylibNames
+, enableManpages ? false
+}:
+
+let
+  self = stdenv.mkDerivation (rec {
+    pname = "clang";
+    inherit version;
+
+    src = runCommand "${pname}-src-${version}" {} ''
+      mkdir -p "$out"
+      cp -r ${monorepoSrc}/cmake "$out"
+      cp -r ${monorepoSrc}/${pname} "$out"
+      cp -r ${monorepoSrc}/clang-tools-extra "$out"
+    '';
+
+    sourceRoot = "${src.name}/${pname}";
+
+    nativeBuildInputs = [ cmake ninja python3 ]
+      ++ lib.optional enableManpages python3.pkgs.sphinx
+      ++ lib.optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames;
+
+    buildInputs = [ libxml2 libllvm ];
+
+    cmakeFlags = [
+      "-DCLANG_INSTALL_PACKAGE_DIR=${placeholder "dev"}/lib/cmake/clang"
+      "-DCMAKE_CXX_FLAGS=-std=c++14"
```

Comment:
question about this flag: why should we add this parameter to force the c++ standard version? Could we use `-DCMAKE_CXX_STANDARD` instead?

Also questioning about whether this is correct. Because clang set its own standard version in `CMakeLists.txt` and the current trunk is **17**.

https://github.com/llvm/llvm-project/search?q=set%28CMAKE_CXX_STANDARD&type=commits

Reply:
Good catch!

As you noted, [clang](https://github.com/llvm/llvm-project/commit/b1356504e63ae821cccf1e051a0d2526bdfef2b0) has transitioned to allowing C++17 code (though this change was made _after_ 15.x branched).

Digging through the `llvmPackages_*` history a bit, it looks like this flag was introduced [way back in the package set for LLVM 3.3](https://github.com/NixOS/nixpkgs/commit/fea2266290d64b11a3bf02bda8764eb6e3d8b7a1#diff-d2c9b97a942fa215c7412a4b809fd9cfebf6c77dc89f6c3132b3516cbaadee91R26) and has been copied over ever since (initially it was added across the package set but today it only remains in `clang`; in [`llvmPackages_9` it was bumped to C++14](https://github.com/NixOS/nixpkgs/blob/20dcc6920f596846aac9787c46a31868f6ecab46/pkgs/development/compilers/llvm/9/clang/default.nix#L30)).

Regardless, it doesn't seem like we need it [anymore](https://github.com/llvm/llvm-project/commit/2724d9e12960cc1d93eeabbfc9aa1bffffa041cc); I'll try removing it from `llvmPackages_15.clang`.

Reply:
https://github.com/NixOS/nixpkgs/pull/194634/commits/fa1ec21011f07a133d0c61313d821c1f103bee18

Still waiting for this to build on all the platforms but I *think* it should be okay.